src/coreclr/tools/aot/Mono.Linker.Tests.Cases/DataFlow/UnresolvedMembers.cs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+
+namespace Mono.Linker.Tests.Cases.DataFlow
+{
+	// NativeAOT will not compile a method with unresolved types in it - it will instead replace it with a throwing method body
+	// So it doesn't produce any of these warnings - which is also correct, because the code at runtime would never get there
+	// it would fail to JIT/run anyway.
+
+	[SkipPeVerify]
+	[SetupLinkerArgument ("--skip-unresolved", "true")]
+	[SetupCompileBefore ("UnresolvedLibrary.dll", new[] { "Dependencies/UnresolvedLibrary.cs" }, removeFromLinkerInput: true)]
+	[ExpectedNoWarnings]
+	class UnresolvedMembers
+	{
+		public static void Main ()
+		{
+			UnresolvedGenericArgument ();
+			UnresolvedAttributeArgument ();
+			UnresolvedAttributePropertyValue ();
+			UnresolvedAttributeFieldValue ();
+			UnresolvedObjectGetType ();
+			UnresolvedMethodParameter ();
+		}
+
+		[Kept]
+		[KeptMember (".ctor()")]
+		class TypeWithUnresolvedGenericArgument<
+			[KeptAttributeAttribute (typeof (DynamicallyAccessedMembersAttribute))]
+		[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicMethods)] T>
+		{
+		}
+
+		[Kept]
+		static void MethodWithUnresolvedGenericArgument<
+			[KeptAttributeAttribute (typeof (DynamicallyAccessedMembersAttribute))]
+		[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicMethods)] T> ()
+		{ }
+
+		[Kept]
+		[ExpectedWarning ("IL2066", "TypeWithUnresolvedGenericArgument", ProducedBy = ProducedBy.Trimmer | ProducedBy.Analyzer)] // Local variable type
```

Comment:
Is there an issue to track this in NativeAOT?

Reply:
See the comment on top of this test file. AOT will actually replace the method bodies with throw and thus won't produce any warnings really. And it's by design - the method would throw when executed (JIT could not compile it), so there's no point in producing warnings for it in AOT.

Technically linker could do the same, but the current behavior of linker is to "ignore but keep" unresolvable references.